src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -6824,7 +6828,15 @@ final void TryProcessFeatureExtAck(boolean featureExtAckReceived) throws SQLServ
 final class TimeoutTimer implements Runnable {
     private final int timeoutSeconds;
     private final TDSCommand command;
-    private Thread timerThread;
+    private volatile Future<?> task;
+    private static final ExecutorService executor = Executors.newCachedThreadPool(new ThreadFactory() {
+        @Override
+        public Thread newThread(Runnable r) {
+            Thread t = Executors.defaultThreadFactory().newThread(r);
```

Comment:
Another way would be to use a dedicated ThreadGroup (for instance with name SQLServerDriver) and to create threads regarding this behaviour.

For instance:
```java
private static final ExecutorService executor = Executors.newCachedThreadPool(new ThreadFactory() {
    final ThreadGroup tg = new ThreadGroup("SQLServerDriver");
    final AtomicInteger counter = new AtomicInteger(0);
    @Override
    public Thread newThread(Runnable r) {
        Thread t = new Thread(tg, "sqlserverdriver-"+counter.incrementAndGet()
        t.setDaemon(true);
        return t;
    }
}
```

Having a dedicated ThreadGroup and well named threads would ease debug in complex software, such as application servers when there are many threads and would help identifying the ones created by the driver.

(While ThreadGroup may be overkill, at least having a good name for threads help a lot)